common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java

```diff
@@ -51,10 +68,16 @@ private void applyProjectPlugin(Project project) {
             CustomBuildScanEnhancements.configureBuildScan(buildScan, project.getGradle());
 
             // Build cache configuration cannot be accessed from a project plugin
+
+            project.afterEvaluate(___ -> {
```

Comment:
I know there is a slight performance penalty for using `afterEvaluate`. Why do we need it here? Could you add a comment to the code?

Reply:
There is a comment for this, but it's within the block. I moved it out.

Reply:
AH ok, i didn't realize that was the reason for this block. But now im curious since the comment is the same, why  do we need to do this in both `settingsEvaluated` and `afterEvaluate`? Would it not be enough to do one or the other? Is this a Gradle 5 vs Gradle 6 thing?

Reply:
Yeah, it's a Gradle 5 vs Gradle 6+ thing.

For Gradle 6+, this is done in the context of settings.gradle(.kts).

For Gradle 5, this is done in the context of a build.gradle(.kts) script, at which point, Settings.settingsEvaluated doesn't really make sense.